File: Data Structures & Algorithms/two-integer-sum/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> s = new Dictionary<int,int>();

        for (int i = 0; i < nums.Length; i++)
        {
            s[nums[i]] = i;
        }

        for (int i = 0; i < nums.Length; i++)
        {
            int temp = target - nums[i];

            if (s.ContainsKey(temp) && i != s[temp])
            {
                return new int[] { i, s[temp] };
            }
        }

        return new int[] { };
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        def dfs(i, arr, total):
            if total == target:
                res.append(arr[:])
                return

            if total > target or i == len(nums):
                return

            arr.append(nums[i])
            dfs(i, arr, total + nums[i])
            arr.pop()

            dfs(i + 1, arr, total)

        dfs(0, [], 0)
        return res